src/encoder.rs

```diff
@@ -479,6 +479,10 @@ pub struct FrameInvariants<T: Pixel> {
   pub pyramid_level: u64,
   pub enable_early_exit: bool,
   pub tx_mode_select: bool,
+  /// If true, this `FrameInvariants` corresponds to a skipped frame and should
+  /// be ignored. Skipped frames occur when a subgop is prematurely ended, for
+  /// example, by a key frame or the end of the video.
+  pub skip: bool,
```

Comment:
This is inelegant at best. I really want to get rid of writing these frameinvariants at all... I guess I'm okay adding this temporarily (it's basically the same as the bool we were returning first)

Also, "skip" is a bitstream feature, so having another concept named skip is confusing. Maybe "dropped" or "invalid" is better.

Reply:
Also, now that this is a fi member, maybe an assert at the beginning of the encode_tile function would make sense to make sure we never try to actually encode one of these frames?

Reply:
Renamed to `invalid`, added an assert in `encode_frame`.